compiler/src/dmd/templatesem.d

```diff
@@ -474,6 +474,11 @@ void templateInstanceSemantic(TemplateInstance tempinst, Scope* sc, ArgumentList
     }
     TemplateDeclaration tempdecl = tempinst.tempdecl.isTemplateDeclaration();
     assert(tempdecl);
+    if (tempdecl.instances is null)
+    {
+        auto inst = new TemplateInstance[TemplateInstanceBox];
+        tempdecl.instances = &inst;
```

Comment:
`tempdecl.instances = inst;

(or maybe `tempdecl.instances = cast(void*)inst; if that doesn't work)

no `&`